### PR TITLE
Some enhancements and added features

### PR DESCRIPTION
**Pull request by gdudek**
Three key features added. 
1) Put spouses (labelled as married) into a subgraph cluster.  This keeps them together in the layout whereas withotu this change spouses can be arbitrarily far apart.
2) Support a few more image types or file suffixes (notably jpg as well as png).
3) For people whose names havea suffix (like Richard III, or Sam Jr) extract that and keep it as part of the name,
4) For people who have a married and maiden name, use both.

### CODE REVIEW

File: ged2dot.py [hunk 1]
```diff
@@ -4,6 +4,12 @@
 #
 # SPDX-License-Identifier: MPL-2.0
 #
+#
+# Modified by Gregory Dudek, April 2, 2024.
```
Comment (vmiklos) [inline ged2dot.py:8]: Best to credit yourself in guide/src/contributors.md instead.

File: ged2dot.py [hunk 2]
```diff
@@ -4,6 +4,12 @@
...
+#  Spouses go in subgraph cluster.
+#  Handle surname suffixes like John Smith Jr.
+#  More image formats supported
```
Comment (vmiklos) [inline ged2dot.py:11]: It would be better to list these in guide/src/news.md instead.

File: go.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+python3 ged2dot.py --familydepth 999 --rootfamily F500001 --input   ../tree_2024_gedcom.ged \
```
Comment (vmiklos) [inline go.sh:1]: It would be better to turn this into tests: see tests/test_ged2dot.py.

**Review by vmiklos — COMMENTED** [review]
Hi!

Thanks for the PR, looks promising. :-) Some suggestions:

1) It may be easier to get this in if you split it to separate PRs. For example the jpg/png support is easy, it could be its own PR, then you would have your first PR in, which means I don't have to approve CI for it. :-)

2) Looks like you haven't run 'make check' after your changes. Please do that. Most importantly, the code has complete statement coverage, so if you add new conditions, you have to write tests for it.

In short, perhaps reduce this PR to just the jpg/png improvement, and do the rest in a follow-up PR? If that works for you, since the first PR is special.

If something is not clear, just ask, I'm happy to help. :-)

**Comment by gdudek** [comment]
Good points. I had originally made the changes for my own use and then decided to do the PR. I'll clean it up as you suggest, but I'll need to find my next open time period even though it's easy.

Greg

**Comment by vmiklos** [comment]
That's fine, this is a hobby project without deadlines. :-)

**Comment by vmiklos** [comment]
Would you mind backing up your current master branch in an other branch, so I can rewrite your master to take a subset that doesn't break CI? Thanks.

**Comment by vmiklos** [comment]
Nevermind, I just restricted this PR to do only 2) to get this PR in, we can get back to the rest in follow-up PRs.

**Comment by vmiklos** [comment]
> Put spouses (labelled as married) into a subgraph cluster. This keeps them together in the layout whereas withotu this change spouses can be arbitrarily far apart.

FWIW, this is a conflicting constraint: either you put the children sorted by their birth date or you group spouses together. But I like the result of this, it's now integrated as https://github.com/vmiklos/ged2dot/commit/2553c855b31828c7b88aef2f248d89cf54e14323

Note that in theory you could still at least order kids at the bottom of the tree who have no kids themselves with something like:

```diff
diff --git a/ged2dot.py b/ged2dot.py
index af14c73..18622dc 100755
--- a/ged2dot.py
+++ b/ged2dot.py
@@ -607,7 +607,7 @@ class DotExport:
                 continue
             family = node
 
-            # Open subgraph of the family.
+            # Open subgraph of the parents.
             cname = "cluster_" + family.get_identifier()
             stream.write(to_bytes("subgraph " + cname + " { style=invis; \n"))
 
@@ -618,12 +618,19 @@ class DotExport:
                 from_husb = family.husb.get_identifier() + " -> " + family.get_identifier() + " [dir=none];\n"
                 stream.write(to_bytes(from_husb))
 
-            # Close subgraph of the family.
+            # Close subgraph of the parents.
             stream.write(to_bytes("}\n"))
 
+            family.child_list.sort(key=lambda child: child.get_config().get_birth())
             for child in family.child_list:
                 stream.write(to_bytes(family.get_identifier() + " -> " + child.get_identifier() + " [dir=none];\n"))
 
+            # Define the order of the children.
+            if len(family.child_list) > 1 and not [c for c in family.child_list if c.fams_ids]:
+                stream.write(to_bytes("{ rank = same; \n"))
+                stream.write(to_bytes(" -> ".join([c.get_identifier() for c in family.child_list])))
+                stream.write(to_bytes(" [ style=invis ];\nrankdir = LR; }\n"))
+
     def store(self, subgraph: List[Node], config: Dict[str, str]) -> None:
         """Exports subgraph to a graphviz path."""
         if config["output"] == "-":
```

But in practice the output I get from e.g. ancestry.com already has these sorted in the right way, so I suggest let's add that complexity when there will be an actual need for it.
